internal/http/client.go

```diff
@@ -385,6 +385,12 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 
 func (c *Client) innerDo(req *Request, errorValue ErrorResponse, i int) (*http.Response, []byte, bool, error) {
 	r, err := req.makeRequest()
+
+	xAccountId := req.request.Context().Value("XAccountID")
+	if xAccountId != nil {
+		req.SetHeader("X-Account-ID", xAccountId.(string))
+	}
```

Comment:
As this is auth logic, I suggest moving this to the [`PersonalAPIKeyCapableV2Authorizer`](https://github.com/newrelic/newrelic-client-go/blob/main/internal/http/auth.go#L29) that is used by [Alerts](https://github.com/newrelic/newrelic-client-go/blob/main/pkg/alerts/alerts.go#L24)), which is called inside of `req.makeRequest()` above.